azurerm/resource_arm_application_gateway.go

```diff
@@ -2645,6 +2712,37 @@ func flattenApplicationGatewayWafConfig(input *network.ApplicationGatewayWebAppl
 		output["rule_set_version"] = *input.RuleSetVersion
 	}
 
+	ruleGroups := make([]interface{}, 0)
+	if disabledRuleGroups := input.DisabledRuleGroups; disabledRuleGroups != nil {
+		for _, ruleGroup := range *disabledRuleGroups {
+			ruleGroupOutput := map[string]interface{}{}
+
+			if ruleGroup.RuleGroupName != nil {
+				ruleGroupOutput["rule_group_name"] = *ruleGroup.RuleGroupName
+			}
+
+			ruleOutputs := make([]interface{}, 0)
+			if rules := ruleGroup.Rules; rules != nil {
+				for _, rule := range *rules {
+					ruleOutputs = append(ruleOutputs, rule)
+				}
+			}
+			ruleGroupOutput["rules"] = ruleOutputs
+
+			ruleGroups = append(ruleGroups, ruleGroupOutput)
+		}
+
+		output["disabled_rule_group"] = ruleGroups
```

Comment:
We should also move the above code into a flatten method that returns an empty `[]interface{}` if input.DisableRuleGroups is nil
```suggestion
		output["disabled_rule_group"] = flattenApplicationGateWayDisabledRuleGroups(input.DisabledRuleGroups)
```